src/Main.kt

```diff
@@ -1,2 +1,20 @@
 fun main() {
-}
\ No newline at end of file
+}
+
+fun fizzBuzz(n: Int): Array<String> {
+    val array = Array(n) {""}
+    var i = 0
+
+    while (i < n) {
```

Comment:
Maybe you could iterate over size of array starting from index 1 to n?
In case n = 5  Then you would have 1,2,3,4,5. 

`  
    for (i in 1..n) {
        when {
            i % 3 == 0 && i % 5 == 0 ->
`